CMakeLists.txt

```diff
@@ -41,6 +41,10 @@ include_directories(${ICAL_INCLUDE_DIRS})
 pkg_check_modules(LIBSECRET REQUIRED libsecret-1)
 include_directories(${LIBSECRET_INCLUDE_DIRS})
 
+# glib
+pkg_check_modules(GLIB REQUIRED glib-2.0)
+include_directories(${GLIB_INCLUDE_DIRS})
+
```

Comment:
glib is an intrinsic dependency of gtk, so you don't need to add it specifically. In fact, if you search the focal code, you should find extant instances of `g_timeout_add_seconds` already.

Reply:
Hmm, I can't think of any reason I'd add the dependency if i didn't run into a problem.  But indeed the code already uses this API, and it doesn't seem to be required.  I *did* have fun indulging in learning and such-- no apologies there but I am sorry for not better cleaning this all up before submitting this PR! Thanks for working with me on this :).

Pretty sure this PR will come down to basically 3 semi-trivial lines! Oh well, less lines of code means less code to maintain ;).

Reply:
> Pretty sure this PR will come down to basically 3 semi-trivial lines

This is of course always the best result